service/controller/clusterapi/v27/key/key.go

```diff
@@ -21,3 +25,26 @@ func ClusterID(cluster v1alpha1.Cluster) string {
 func ClusterNamespace(cluster v1alpha1.Cluster) string {
 	return ClusterID(cluster)
 }
+
+func DockerVolumeResourceName(cluster v1alpha1.Cluster) string {
+	return getResourcenameWithTimeHash("DockerVolume", cluster)
+}
+
+func MasterInstanceResourceName(cluster v1alpha1.Cluster) string {
+	return getResourcenameWithTimeHash("MasterInstance", cluster)
+}
+
+// getResourcenameWithTimeHash returns a string cromprised of some prefix, a
+// time hash and a cluster ID.
+func getResourcenameWithTimeHash(prefix string, cluster v1alpha1.Cluster) string {
+	id := strings.Replace(ClusterID(cluster), "-", "", -1)
+
+	h := sha1.New()
+	h.Write([]byte(strconv.FormatInt(time.Now().UnixNano(), 10)))
+	timeHash := fmt.Sprintf("%x", h.Sum(nil))[0:5]
```

Comment:
I wonder how often this would collide? If one uses only last 5 digits of seconds since epoch, it would be roughly every 27th hour. This one here is slightly more random, but I think all the same.

Reply:
The cluster ID is part of it as well and we are not updating the same cluster within the same second multiple times.

Reply:
We use these functions for generating CF template resource names. The timestamp determines if a resource should be changed by CF or not. This is the reason why we have to provide the old name when scaling a cluster.

Reply:
Ok. Any specific reason then for SHA1 instead of just using seconds as such and taking 5 least significant ones from that?

Reply:
I just moved the code Tuomas. That exists like that already. Reason might just be not having thought too much about it. I cannot tell you more at this point. Let's maybe hang when there are more concerns.